### PR TITLE
Feature/kc strongly typed

### PR DESCRIPTION
**Pull request by KaneCheshire**
- Removes dependency on the `XCTest+Gherkin extension`
- Adds new Step and Scenario structs
- Set project to use consistent tabs vs spaces (so diffs show up correctly regardless of the devs Xcode settings)
- Renamed `UITestPage` to `UITestScreen`
- Removed the global functions and moved to the `UITestScreen` extension

Still todo: update the readme/docs

### CODE REVIEW

File: TABTestKit/Classes/TestBase.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ open class TestBase: XCTestCase {
     launchApp()
   }
   
-  /// Launches the application with options if needed
+  /// Launches the application with options if needed // TODO: What does this mean "with options if needed"?
```
Comment (zacoid55) [inline TABTestKit/Classes/TestBase.swift]: I fear this is a remnant, but there was a `launchAppWIthArguments()` or something similar at one point

File: TABTestKit/Classes/UITestScreen.swift [hunk 2]
```diff
@@ -49,9 +58,8 @@ open class GlobalFunctions {
   ///   - xCoordinate: The given coordinate across the x axis
   ///   - yCoordinate: The given coordinate down the y axis
   func tapCoordinate(x xCoordinate: Double, y yCoordinate: Double) {
-    //Sets up a base coordinate (the top left of the screen)
-    let normalized = App.shared.coordinate(withNormalizedOffset: CGVector(dx: 0, dy: 0))
-    let coordinate = normalized.withOffset(CGVector(dx: xCoordinate, dy: yCoordinate))
+    let topLeftOfScreen = App.shared.coordinate(withNormalizedOffset: CGVector(dx: 0, dy: 0))
+    let coordinate = topLeftOfScreen.withOffset(CGVector(dx: xCoordinate, dy: yCoordinate))
```
Comment (zacoid55) [inline TABTestKit/Classes/UITestScreen.swift]: Ah, much better - thank you

File: TABTestKit/Classes/Step.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+/// Defines an executable step to be used in tests.
+///
+/// Steps that are initialised with a function that doesn't take arguments can omit the parenthesis,
+/// i.e: Given(I: logIn) instead of Given(I: logIn())
+/// To learn more about how to use a Step read this article: https://edit.theappbusiness.com/swifty-gherkins-part-1-28abba7dfd8
+public struct Step {
+  
+  @discardableResult
+  public init(_ handler: () -> Void, function: StaticString = #function) {
```
Comment (KaneCheshire) [inline TABTestKit/Classes/Step.swift]: I don't think these `function`s are needed now

**Comment by KaneCheshire** [comment]
I'll merge this when I know I've got time to start the README updates so that they're not contradicting each other for too long

**Pull request merged**